web/modules/custom/dpl_protected_nodes/dpl_protected_nodes.module

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/**
+ * @file
+ * Primary module hooks for DPL Protected Nodes module.
+ */
+
+use Drupal\Core\Entity\EntityFormInterface;
+use Drupal\Core\Form\FormStateInterface;
+use function Safe\preg_match;
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ *
+ * @throws \Safe\Exceptions\PcreException
+ */
+function dpl_protected_nodes_form_node_confirm_form_alter(array &$form, FormStateInterface $form_state): void {
+  if ($form_state->getFormObject() instanceof EntityFormInterface) {
+    /** @var \Drupal\node\NodeInterface $node */
+    $node = $form_state->getFormObject()->getEntity();
+    $node = [$node->id() => $node->getTitle()];
+
+    $form = dpl_protected_nodes_is_nodes_protected($node, $form);
+  }
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function dpl_protected_nodes_form_node_delete_multiple_confirm_form_alter(array &$form, FormStateInterface $form_state): void {
+  $items = $form['entities']['#items'];
+  $nodes = [];
+
+  foreach ($items as $key => $value) {
+    $key = explode(':', $key, 2);
+    $nodes[$key[0]] = $value;
+  }
+
+  $form = dpl_protected_nodes_is_nodes_protected($nodes, $form);
+}
+
+/**
+ * Checks if node is protected and alters the delete confirm form accordingly.
+ *
+ * @param mixed[] $nodes
+ *   The node id to check.
+ * @param mixed[] $form
+ *   The form array to alter.
+ *
+ * @return mixed[]
+ *   The altered form array.
+ */
```

Comment:
@kasperg Made me aware of that since we are type hinting the function parameters and defining the return types of the functions  we do not need to replicate the documention of those in the dockblock. Does it make sense?

Reply:
Hmm, yes and no. I thought it was standard to add docblocks for all functions. If you don't and you omit @params, @return etc. then phpstan is complaining about it? Or am i misinterpreting what you meant?

Reply:
No. We have a conflict between accepting no params/return in docblocks and a phpstan rule that wants a definition of the array type. I guess we just let params/returns with `mixed[]` be as they are...